crates/hotshot/src/traits/networking/libp2p_network.rs

```diff
@@ -662,35 +686,19 @@ impl<K: SignatureKey + 'static> Libp2pNetwork<K> {
                     handle.subscribe("DA".to_string()).await.unwrap();
                 }
 
-                // we want our records published before
-                // we begin participating in consensus
-                //
-                // Note: this serialization should never fail,
-                // and if it does the error is unrecoverable.
-                while handle
-                    .put_record(
-                        &bincode::serialize(&pk).unwrap(),
-                        &bincode::serialize(&handle.peer_id()).unwrap(),
-                    )
-                    .await
-                    .is_err()
-                {
-                    async_sleep(Duration::from_secs(1)).await;
-                }
-
+                // Map our staking key to our Libp2p Peer ID so we can properly
+                // route direct messages
                 while handle
                     .put_record(
-                        &bincode::serialize(&handle.peer_id()).unwrap(),
-                        &bincode::serialize(&pk).unwrap(),
```

Comment:
Do you know why we had this reverse mapping in the DHT? Seems like it's fine to get rid of

Reply:
No idea, which is why I got rid of it. We only ever use the DHT for the routing + lookup in the other direction